feature: `make crds` should include APIExport generation

### Feature Description

edge-mc should have the functional equivalent of https://github.com/kcp-dev/kcp/blob/cb8dd3c46470f30414c6789be956162ccb21f669/hack/update-codegen-crds.sh#L55-L58 --- ideally without maintaining a copy of the apigen tool.

### Proposed Solution

Have the Makefile or a script `go install github.com/kcp-dev/kcp/cmd/apigen@$right_version` into some tools binary directory and invoke it from there.  https://github.com/kcp-dev/kcp/pull/2669 has merged and this makes it possible to `go install` apigen.

I just tested the apigen from kcp commit cb8dd3c46470f30414c6789be956162ccb21f669 and it produced exactly the same output as the apigen binary that I built earlier from kcp commit 4506fdc06406 (which is the one that edge-mc currently depends on).

### Alternative Solutions

- Maintain a copy of the apigen source.
- Have the Makefile or script `git clone` kcp to get the source and then build it.

### Want to contribute?

- [x] I would like to work on this issue.

### Additional Context

_No response_

I was able to get apigen installed as part of the makefile tools section.  apigen is now installed in the /hack/tools path.  Now I am testing update of crds with apigen

from within update-codegen-crds.sh I am running:

go run "${REPO_ROOT}"/hack/tools/apigen --input-dir "${REPO_ROOT}"/config/crds --output-dir "${REPO_ROOT}"/config/root-phase0

this yields:

  make crds codegen         
./hack/update-codegen-crds.sh
+ [[ -z hack/tools/controller-gen-v0.10.0 ]]
+++ dirname ./hack/update-codegen-crds.sh
++ cd ./hack/..
++ pwd
+ REPO_ROOT=/Users/andan02/projects/edge-mc
+ cd /Users/andan02/projects/edge-mc/pkg/apis
+ ../../hack/tools/controller-gen-v0.10.0 crd rbac:roleName=manager-role webhook paths=./... output:crd:artifacts:config=/Users/andan02/projects/edge-mcconfig/crds
+ for CRD in '"${REPO_ROOT}"/config/crds/*.yaml'
+ '[' -f /Users/andan02/projects/edge-mc/config/crds/edge.kcp.io_edgeplacements.yaml-patch ']'
+ for CRD in '"${REPO_ROOT}"/config/crds/*.yaml'
+ '[' -f /Users/andan02/projects/edge-mc/config/crds/edge.kcp.io_singleplacementslice.yaml-patch ']'
+ go run /Users/andan02/projects/edge-mc/hack/tools/apigen --input-dir /Users/andan02/projects/edge-mc/config/crds --output-dir /Users/andan02/projects/edge-mc/config/root-phase0
go: no packages loaded from /Users/andan02/projects/edge-mc/hack/tools/apigen
make: *** [crds] Error 1

please advise

--- a/Makefile
+++ b/Makefile
@@ -180,7 +180,7 @@ $(YAML_PATCH):
 $(GOTESTSUM):
 	GOBIN=$(TOOLS_GOBIN_DIR) $(GO_INSTALL) gotest.tools/gotestsum $(GOTESTSUM_BIN) $(GOTESTSUM_VER)
 
-crds: $(CONTROLLER_GEN) $(YAML_PATCH)
+crds: $(CONTROLLER_GEN) $(API_GEN) $(YAML_PATCH)
 	./hack/update-codegen-crds.sh
 .PHONY: crds
 

--- a/hack/update-codegen-crds.sh
+++ b/hack/update-codegen-crds.sh
@@ -54,4 +54,4 @@ done
 #     paths="${REPO_ROOT}/test/e2e/reconciler/cluster/..." \
 #     output:crd:artifacts:config="${REPO_ROOT}"/test/e2e/reconciler/cluster/
 
-"${API_GEN}" --input-dir "${REPO_ROOT}"/config/crds --output-dir "${REPO_ROOT}"/config/exports
+"${REPO_ROOT}/${API_GEN}" --input-dir "${REPO_ROOT}"/config/crds --output-dir "${REPO_ROOT}"/config/exports

